core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,100 @@
-      return UrlInterpolationService.interpolateUrl(
-        RESOLVE_ISSUE_URL, {
-          exploration_id: explorationId
-        });
-    };
-    return {
-      fetchIssues: function(explorationId, explorationVersion) {
-        if (cachedIssues !== null) {
-          return $q.resolve(cachedIssues);
-        } else {
-          return $http.get(getFullIssuesUrl(explorationId), {
-            params: {
-              exp_version: explorationVersion
-            }
-          }).then(function(response) {
-            var unresolvedIssueBackendDicts = response.data;
-            cachedIssues = unresolvedIssueBackendDicts.map(
-              PlaythroughIssueObjectFactory.createFromBackendDict);
-            return cachedIssues;
-          });
-        }
-      },
-      fetchPlaythrough: function(expId, playthroughId) {
-        return $http.get(getFullPlaythroughUrl(expId, playthroughId)).then(
-          function(response) {
-            var playthroughBackendDict = response.data;
-            return PlaythroughObjectFactory.createFromBackendDict(
-              playthroughBackendDict);
-          });
-      },
-      resolveIssue: function(issueToResolve, expId, expVersion) {
-        return $http.post(getFullResolveIssueUrl(expId), {
-          exp_issue_dict: issueToResolve.toBackendDict(),
-          exp_version: expVersion
-        }).then(function() {
-          var issueIndex = cachedIssues !== null ?
-            cachedIssues.findIndex(function(issue) {
-              return angular.equals(issue, issueToResolve);
-            }) : -1;
-          if (issueIndex === -1) {
-            var invalidIssueError = new Error(
-              'An issue which was not fetched from the backend has been ' +
-              'resolved');
-            return $q.reject(invalidIssueError);
-          } else {
-            cachedIssues.splice(issueIndex, 1);
-          }
-        });
-      },
-    };
-  }]);
+export class PlaythroughIssuesBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private urlInterpolationService: UrlInterpolationService,
+    private playthroughIssueObjectFactory: PlaythroughIssueObjectFactory) { }
+
+  private cachedIssues = null;
+
+  private getFullIssuesUrl(explorationId: string) {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_ISSUES_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  private getFullPlaythroughUrl(expId: string, playthroughId: string) {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.FETCH_PLAYTHROUGH_URL, {
+        exploration_id: expId,
+        playthrough_id: playthroughId
+      });
+  }
+
+  private getFullResolveIssueUrl(explorationId: string) {
+    return this.urlInterpolationService.interpolateUrl(
+      ServicesConstants.RESOLVE_ISSUE_URL, {
+        exploration_id: explorationId
+      });
+  }
+
+  fetchIssues(explorationId: string, explorationVersion: any) {
```

Comment:
explorationVersion should be `string` right? Please check and remove `any`.

Reply:
I'm confused over here, if I set it as string the spec file starts throwing error as over there fetchIssues is called using a string and a number parameter.

If I change the function declaration to string, number as parameter types to suit the spec file, the get call params inside the function, this -> params: { exp_version: explorationVersion }, starts throwing error, thats why I gave it a any parameter type.

Reply:
@bansalnitish please guide me on this issue

Reply:
Hey @vinayakvvg, Thanks for pinging me again -- I think I missed this one.

So, what I did is I looked at the spec file and the line that calls this function. Example this line:
```
playthroughIssuesBackendApiService.fetchIssues('7', 1).then(
  successHandler, failureHandler);
```
which clearly states that first argument is string and second should be number. Could you please change the `explorationVersion` to `number` now?

Reply:
@bansalnitish, hi when I assign number to explorationVersion. It gives the following error while running the typescript_checks. 
--->

The last overload gave the following error.
    Type 'number' is not assignable to type 'string | string[]'.

Reply:
I checked this and you're right here. Let's add a comment on top of this something like
```
TODO(#7165): This has been marked any since marking explorationVersion to number throws an error "Type 'number' is not assignable to type 'string | string[]'" and if this is marked to string it throws an error "TS2345: Argument of type '1' is not assignable to parameter of type 'string'".
```

Reply:
@bansalnitish 
Okay, adding the comment.
I'll be raising a PR after this.

Please suggest me some other issue to work on as well.

Reply:
@bansalnitish 
Okay, adding the comment.
I'll be raising a PR after this.

Please suggest me some other issue to work on as well.

Reply:
I'd say taken any other service from the issue #8472 itself. There are a lot of services mentioned in the tracker sheet attached there. Thanks!

Reply:
Okay, after selecting a file from tracker, i can just post on this link #8472 to assign that to me right ?

Reply:
Assigned!